README.md

```diff
@@ -29,19 +20,49 @@ go get github.com/ipfs/go-ipld-git
 ```
 
 ## About
+
 This is an IPLD codec which handles git objects. Objects are transformed
-into IPLD graph in the following way:
+into IPLD graph as detailed below. Objects are demonstrated here using both
+[IPLD Schemas](https://ipld.io/docs/schemas/) and example JSON forms.
+
+### Commit
+
+```ipldsch
+type GpgSig string
+
+type PersonInfo struct {
+  date String
+  timezone String
+  email String
+  name String
+}
+
+type Commit struct {
+  author optional PersonInfo
+  committer optional PersonInfo
+  message String
+  parents [&Commit]
+  tree &Tree # see "Tree" section below
+  encoding optional String
+  signature optional GpgSig
+  mergeTag [Tag]
+  other [String]
+}
+```
+
+As JSON, real data would look something like:
 
-* Commit:
 ```json
 {
   "author": {
-    "date": "1503667703 +0200",
+    "date": "1503667703",
+    "timezone": "+0200",
     "email": "author@mail",
     "name": "Author Name"
   },
   "committer": {
-    "date": "1503667703 +0200",
+    "date": "1503667703",
+    "timezone": "+0200",
```

Comment:
(delayed reply to this) the main idea here is to keep the data lossless and not do too many tricks to make it palatable to the user.

Encoded form looks something like:

```
author A U Thor <author@example.com> 1465981137 +0000
```

But there's even some flexibility in what those final two fields can contain and apparently whether we get two, one or zero of them (I'm not sure about that detail though). So we treat them as strings and present them to the user as they are.

The current version of this codec keeps them internally but only presents a nice human-readable form as a DAG node, which isn't going to work when we're treating DAG traversal as traversal over the data model like we're doing with ipld-prime.

We did discuss (in this thread and other places) some options for making the human-readable form _emerge_ out of the data to be available, but @willscott made the case that since this codec is a bit of a reference codec that we'd want to point others to for how to implement one that we'd better keep it simple for now.

So, lossless data, pure data model, no trickery, user gets to interpret the nodes as they want.